### PR TITLE
Fixed some build errors for Windows target

### PR DESCRIPTION
**Pull request by GunGraveKoga**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -323,6 +323,7 @@ add_swift_library(Foundation
                     ${CoreFoundation_INTERFACE_LIBRARIES}
                     ${WORKAROUND_SR9138}
                     ${WORKAROUND_SR9995}
+                    -Xlinker;/FORCE
```
Comment (compnerd) [inline CMakeLists.txt]: This is wrong.  We shouldn't be using `/FORCE`.  Furthermore, this needs to be specific to Windows.  What symbol is being defined multiply?  Those should have all been fixed by now.

Comment (GunGraveKoga) [reply CMakeLists.txt]: I squash two latest commits.
Right now I cannot list symbols which causing LNK1169 error cos hasnot access to Win32 machine. I will try to rebuild Foundation without /FORCE flag later today and attach build-log

**Comment by compnerd** [comment]
Great!  I look forward to the build log - I've been building this for quite some time (a couple of months) and I've not seen any multiply defined symbols since I fixed the underlying issue in the swift compiler.

**Comment by GunGraveKoga** [comment]
Build log 
[foundation.log](https://github.com/apple/swift-corelibs-foundation/files/3005737/foundation.log)
Multiply defined symbols:
Process.swift: $sSo7in_addrV20__Unnamed_union_S_unVwet (priviously defined in Host.swift)
Process.swift: $sSo7in_addrVwst (priviously defined in Host.swift)
MultiHandle.swift: $sSo20CFURLSessionEasyCodeVwst (priviously defined in EasyHandle.swift)

Used toolchain: https://dev.azure.com/compnerd/3133d6ab-80a8-4996-ac4f-03df25cd3224/_apis/build/builds/757/artifacts?artifactName=toolchain&api-version=5.1-preview.5&%24format=zip (downloaded 23.05.2019 after 14:00 MSK)

**Comment by compnerd** [comment]
Hmm, this is new, and interesting!  I've not seen this yet, but I have a feeling that this might be related to recent compiler changes.  This is a bug on the swift side.

```
$sSo7in_addrV20__Unnamed_union_S_unVwet ---> getEnumTagSinglePayload value witness for __C.in_addr.__Unnamed_union_S_un
$sSo7in_addrVwst ---> storeEnumTagSinglePayload value witness for __C.in_addr
$sSo20CFURLSessionEasyCodeVwst ---> storeEnumTagSinglePayload value witness for __C.CFURLSessionEasyCode
```

Seems related to the enumeration payload accessors.  Could you please file a SR for me?

**Comment by compnerd** [comment]
I just did a clean build, and I cannot seem to reproduce this duplicate definition.  How are you building Foundation?

**Comment by GunGraveKoga** [comment]
I have toolchain downloaded from your build-bot https://dev.azure.com/compnerd/windows-swift and I invoked cmake configure like: ```cmake -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=clang-cl -DCMAKE_SWIFT_COMPILER=swiftc -DCURL_LIBRARY=C:/Proj/swift-msvc/curl/builds/libcurl-vc15-x64-release-static-ipv6-sspi-winssl/lib/libcurl_a.lib -DCURL_INCLUDE_DIR=C:/Proj/swift-msvc/curl/builds/libcurl-vc15-x64-release-static-ipv6-sspi-winssl/include -DENABLE_TESTING=NO -DICU_ROOT=C:/Library/icu63_1 -DLIBXML2_LIBRARY=C:/Proj/swift-msvc/libxml2/win32/bin.msvc/libxml2_a.lib -DLIBXML2_INCLUDE_DIR=C:/Proj/swift-msvc/libxml2/include -DFOUNDATION_PATH_TO_LIBDISPATCH_SOURCE=C:/Proj/swift-msvc/swift-corelibs-dispatch -DFOUNDATION_PATH_TO_LIBDISPATCH_BUILD=C:/Proj/swift-msvc/libdispatch -DCMAKE_INSTALL_PREFIX=C:/proj/swift-msvc/install C:/Proj/swift-msvc/swift-corelibs-foundation```

I will try to use newly toolchain builds to reproduce issue

**Comment by GunGraveKoga** [comment]
Nope, same result
```bash
C:\Proj\swift-msvc\foundation>swiftc --version
Swift version 5.0-dev (LLVM 23d88de9b5, Swift 8420a00ed8)
Target: x86_64-unknown-windows-msvc
```

**Comment by compnerd** [comment]
Interesting, I'll try with the nightly builds as I'm usually building with a local build.

**Comment by compnerd** [comment]
Hmm, seems to reproduce with Release but not RelWithDebInfo?  Seems like we aren't marking something as COMDAT.

**Comment by compnerd** [comment]
SR-10192

**Comment by GunGraveKoga** [comment]
I did remove commit with linker /FORCE flag. Now this pull request fix only preconditionFailure in NSTemporaryDirectory() function on Windows

**Comment by compnerd** [comment]
@swift-ci please test and merge

**Comment by compnerd** [comment]
@swift-ci please test and merge

**Pull request merged**